dataflow/src/org/checkerframework/dataflow/cfg/node/LambdaResultExpressionNode.java

```diff
@@ -0,0 +1,81 @@
+package org.checkerframework.dataflow.cfg.node;
+
+import com.sun.source.tree.ExpressionTree;
+import java.util.Collection;
+import java.util.Collections;
+import javax.lang.model.util.Types;
+import org.checkerframework.dataflow.util.HashCodeUtils;
+import org.checkerframework.javacutil.InternalUtils;
+
+/*>>>
+import org.checkerframework.checker.nullness.qual.Nullable;
+*/
+
+/**
+ * A node for the single expression body of a single expression lambda
+ *
+ * @author Lazaro Clapp
+ */
+public class LambdaResultExpressionNode extends Node {
+
+    protected ExpressionTree tree;
+    protected /*@Nullable*/ Node result;
+
+    public LambdaResultExpressionNode(
+            ExpressionTree t, /*@Nullable*/ Node result, Types types /*,
+            LambdaExpressionTree lambda,
+            Symbol.MethodSymbol methodSymbol*/) {
+        super(InternalUtils.typeOf(t));
+        this.result = result;
+        tree = t;
+        //result.setAssignmentContext(new AssignmentContext.LambdaReturnContext(methodSymbol));
```

Comment:
Did you run into any problems with the assignment context? Does this require more work or can this comment be removed?

Reply:
Nope. Removing, this is left-over code from another approach at this.